File: Languages/Java/Insertion_Sort.java
import java.io.*;
import java.lang.*;
import java.util.*;

class Insertion_Sort { 
    void sort(int arr[]) 
    { 
        int n = arr.length; 
        for (int i = 1; i < n; ++i) { 
            int key = arr[i]; 
            int j = i - 1; 
            while (j >= 0 && arr[j] > key) { 
                arr[j + 1] = arr[j]; 
                j = j - 1; 
            } 
            arr[j + 1] = key; 
        } 
    } 
    static void printArray(int arr[]) 
    { 
        int n = arr.length; 
        for (int i = 0; i < n; ++i) 
            System.out.print(arr[i] + " "); 
  
        System.out.println(); 
    } 
    
    public static void main(String args[]) 
    {   
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter the length of array");
        int N=sc.nextInt();
        int arr[] = new int[N]; 
        System.out.println("Enter the values of array");
        for(int i=0;i<N;i++){
          arr[i]=sc.nextInt();
        }
        Insertion_Sort ob = new Insertion_Sort(); 
        ob.sort(arr); 
  
        printArray(arr); 
    } 
}

File: Languages/C++/insertion_sort.cpp
#include <iostream>
using namespace std;

class insertionSort {
  public:
  int arr[50], n, i, j, temp;
  void get_ele() {
    cout << "Enter the number of elements: ";
    cin >> n;
    for (i = 0; i < n; i++) {
      cout << "Enter number: ";
      cin >> arr[i];
    }
  }
  void find() {
    for (i = 1; i < n; i++) {
      temp = arr[i];
      j = i - 1;
      while (temp < arr[j] && j >= 0) {
        arr[j + 1] = arr[j];
        --j;
      }
      arr[j + 1] = temp;
    }
    cout << "\nSorted Elements are: " << endl;
    for (i = 0; i < n; i++) {
      cout << arr[i] << " ";
    }
  }
};
int main() {
  insertionSort obj;
  obj.get_ele();
  obj.find();
}